src/PubSubClient.cpp

```diff
@@ -498,19 +498,18 @@ bool PubSubClient::publish_P(const char* topic, const uint8_t* payload, size_t p
 bool PubSubClient::beginPublish(const char* topic, size_t plength, bool retained) {
     if (connected()) {
         // Send the header and variable length field
-        size_t length = MQTT_MAX_HEADER_SIZE;
-        length = writeString(topic, this->buffer, length);
+        size_t topicLen = writeString(topic, this->buffer, MQTT_MAX_HEADER_SIZE) - MQTT_MAX_HEADER_SIZE;
         const uint8_t header = MQTTPUBLISH | (retained ? MQTTRETAINED : 0);
```

Comment:
This isn't a `header`, but rather a `msgType`

Reply:
It's byte 1 of the header ...
https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718038